aurt/rigid_body_dynamics.py

```diff
@@ -429,28 +415,28 @@ def regressor_joint_parameters_for_joint(self, j, par_j):
         return res
 
     def regressor_joint(self, j):
-        return cache_object(RigidBodyDynamics.filepath_regressor_joint(j+1), lambda: self.regressor()[j, :])
+        return self._cache.get_or_cache(RigidBodyDynamics.filepath_regressor_joint(j+1), lambda: self.regressor()[j, :])
 
     def regressor(self, output_filename="rigid_body_dynamics_regressor"):
         """
         The regressor matrix formulated in terms of the Base Inertial Parameters (BIP).
         """
-        filepath_regressor = from_cache(output_filename)
+        return self._cache.get_or_cache(output_filename, self.compute_regressor)
 
-        def compute_regressor():
-            regressor_sip_exist = self._regressor_sip_exist_instantiated_dh()
-            args_sym = self.q[1:] + self.qd[1:] + self.qdd[1:] + self._g.T.tolist()[0]  # list concatenation
-            sys.setrecursionlimit(int(1e6))  # Prevents errors in sympy lambdify
-            regressor_sip_exist_func = sp.lambdify(args_sym, regressor_sip_exist, 'numpy')
+    def compute_regressor(self):

```

Comment:
should be non exposed, i.e. have a prefix underscore

Reply:
i don't get why this function should not be defined inside the scope of "regressor()". It is only used inside this function.

Reply:
I put it out thinking of our future experiments where we may want to call it directly to do other things with it.